lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -527,12 +529,12 @@ defmodule ExUnit.Assertions do
     do: raise(ArgumentError, "timeout must be a non-negative integer, got: #{inspect(timeout)}")
 
   @doc false
-  def __timeout__(binary, pins, pattern_finder, timeout) do
+  def __timeout__(pattern, code, pins, pattern_finder, timeout) do
     {:messages, messages} = Process.info(self(), :messages)
 
     if Enum.any?(messages, pattern_finder) do
       """
-      Found message matching #{binary} after #{timeout}ms.
+      Found message matching #{Macro.to_string(pattern)} after #{timeout}ms.
```

Comment:
We should probably raise on both branches then? Otherwise it is weird one raises and the other returns the error message.